src/native/libs/System.Globalization.Native/pal_localeStringData.c

```diff
@@ -279,26 +286,50 @@ int32_t GlobalizationNative_GetLocaleInfoString(const UChar* localeName,
             }
             break;
         case LocaleString_ThousandSeparator:
-            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_GROUPING_SEPARATOR_SYMBOL, value, valueLength);
+            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_GROUPING_SEPARATOR_SYMBOL, value, valueLength, NULL);
             break;
         case LocaleString_DecimalSeparator:
-            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_DECIMAL_SEPARATOR_SYMBOL, value, valueLength);
+            status = GetLocaleInfoDecimalFormatSymbol(locale, UNUM_DECIMAL_SEPARATOR_SYMBOL, value, valueLength, NULL);
             break;
         case LocaleString_Digits:
-            status = GetDigitSymbol(locale, status, UNUM_ZERO_DIGIT_SYMBOL, 0, value, valueLength);
-            // symbols UNUM_ONE_DIGIT to UNUM_NINE_DIGIT are contiguous
-            for (int32_t symbol = UNUM_ONE_DIGIT_SYMBOL; symbol <= UNUM_NINE_DIGIT_SYMBOL; symbol++)
             {
-                int charIndex = symbol - UNUM_ONE_DIGIT_SYMBOL + 1;
-                status = GetDigitSymbol(
-                    locale, status, (UNumberFormatSymbol)symbol, charIndex, value, valueLength);
+                // Native digit can be more than one character (e.g. ccp-Cakm-BD locale which using surrogate pairs to represent the native digit).
+                // We'll separate the native digits in the returned buffer by the character '\uFFFF'.
+                int charIndex = 0;
+                int symbolLength = 0;
+
+                status = GetDigitSymbol(locale, status, UNUM_ZERO_DIGIT_SYMBOL, 0, value, valueLength, &symbolLength);
+                charIndex += symbolLength;
+
+                if (U_SUCCESS(status) && (uint32_t)charIndex < (uint32_t)valueLength)
+                {
+                    value[charIndex++] = 0xFFFF;
+
+                    // symbols UNUM_ONE_DIGIT to UNUM_NINE_DIGIT are contiguous
+                    for (int32_t symbol = UNUM_ONE_DIGIT_SYMBOL; symbol <= UNUM_NINE_DIGIT_SYMBOL && charIndex < valueLength - 3; symbol++)
```

Comment:
Returning partial data will be treated as invalid data because we'll not have the 10 digits in the buffer. You can see that on the managed side. It is very unlikely we can run out of space here because we are using a buffer of 100 characters which is more than enough. we always don't fail in general when we cannot read the data and we fall back to the ASCII digits at that time.

Reply:
It looks like we're assuming that a symbol has a maximum length of 2?  Is that guaranteed by ICU?

Reply:
This is the case today that max digit string length is 2. Even in the future if this increased (which is very unlikely), we will still be ok because we are using a much bigger buffer (100-character length) and we shouldn't run into any edge cases.

Reply:
Might be worth an assert or two.

Reply:
I have added the assert in the managed side to catch any case we fail reading the native digits successfully from ICU.